homeassistant/components/http.py

```diff
@@ -271,6 +326,9 @@ def _handle_request(self, method):  # pylint: disable=too-many-branches
                     "API password missing or incorrect.", HTTP_UNAUTHORIZED)
 
             else:
```

Comment:
`else:` should be `elif require_auth:`. Otherwise we will also write a session if `require_auth == False`